lib/bech32/src/Codec/Binary/Bech32/Internal.hs

```diff
@@ -90,7 +95,45 @@ import qualified Data.ByteString as BS
 import qualified Data.ByteString.Char8 as B8
 
 {-------------------------------------------------------------------------------
-                          Human Readable Parts
+                                 Data Part
+-------------------------------------------------------------------------------}
+
+-- | Represents the data part of a Bech32 string, as defined here:
+--   https://git.io/fj8FS
+newtype DataPart = DataPart
+    { getDataPart :: [Word5] }
+    deriving Eq
+
+instance Monoid DataPart where
+    mempty = DataPart []
+
+instance Semigroup DataPart where
+    (DataPart d0) <> (DataPart d1) = DataPart $ d0 <> d1
+
+instance Show DataPart where
+    show (DataPart dp) = "DataPart " <> show ((word5ToChar Arr.!) <$> dp)
+
+-- | Constructs a 'DataPart' from a 'ByteString'.
+--
+-- This function encodes a 'ByteString' in such a way that guarantees it can be
+-- successfully decoded with the 'dataPartToBytes' function:
+--
+-- > dataPartToBytes (dataPartFromBytes b) == Just b
```

Comment:
Minor remark but I believe the documentation annotation here should start with a `>>>` otherwise haddock won't display this as you expect ^.^

Reply:
According to the [Haddock docs](https://www.haskell.org/haddock/doc/html/ch03s08.html), both styles are possible. With a single `>` we get:

![Screenshot_2019-05-24 Codec Binary Bech32 Internal](https://user-images.githubusercontent.com/206319/58296556-c0240380-7e06-11e9-94e5-20a777ce02d5.png)

If we instead use `>>>`, then we get this:

![Screenshot_2019-05-24 Codec Binary Bech32 Internal](https://user-images.githubusercontent.com/206319/58296653-1abd5f80-7e07-11e9-8dd4-2ad909345f05.png)

I think the latter style is provided for giving examples.